trieste/utils/misc.py

```diff
@@ -55,3 +57,112 @@ def to_numpy(t: TensorType) -> np.ndarray:
         return t.numpy()
 
     return t
+
+
+T_co = TypeVar("T_co", covariant=True)
+""" An unbounded covariant type variable. """
+
+
+class Result(Generic[T_co], ABC):
```

Comment:
@st-- 's qu: how does this compare to the `result` package ?

Reply:
i think [returns](https://github.com/dry-python/returns) is the best candidate for this, but I'd like to hold off adding that dependency until we need more functionality from it. For now, we don't need much.